### DISCUSSION TITLE
Can I assume that the DANDI asset ID is guaranteed to always point to the same file content?

### DISCUSSION THREAD

**Discussion by magland** [dandi/helpdesk#111] [General]
Can I assume that the DANDI asset ID is guaranteed to always point to the same file content? In my projects, I would like to switch to using the DANDI API download URLs rather than the s3 bucket URLs in order to be able to support embargoed dandisets. However, it's important to me that the file content for a given URL cannot change over time.

This is related to the following neurosift/dandi-archive issues:
https://github.com/dandi/dandi-archive/pull/1690
https://github.com/flatironinstitute/neurosift/issues/112

**Comment by satra** [comment]
a given asset id will always have the same file content, but the same file content may have different asset ids.  asset = metadata + file, and the id changes if any bit of either component changes.
